src/main/java/payapal/Perfection.java

```diff
@@ -0,0 +1,42 @@
+package main.java.payapal;
+
+public class Perfection {
+
+    private static Perfection perf;
+
+    public synchronized static Perfection getPerf() {
+        if (perf == null) {
+            perf = new Perfection();
+        }
+        return perf;
+    }
+
+
+    public static boolean isPerfect(long candidate) {
+        boolean retVal;
+        long[] divisors = GetDivisors(candidate);
+        int sum = 0;
+        for (int d = 0; d < 1000; d++)
```

Comment:
We are not sure that 1000 is going to be the number of divisors, that depends on the candidate number. I recommend instead of using an array to use a collection (List probably more appropriate) cause arrays are fixed in length and also with the collection we can invoke as a limit the size() method. Something like:
```
for (int d = 0; d < divisors.size(); d++)
```